tests/base/symmetric_tensor_36.cc

```diff
@@ -0,0 +1,96 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+// Check the tensor copy constructors for different number types
+
+#include "../tests.h"
+#include <deal.II/base/symmetric_tensor.h>
+#include <deal.II/base/tensor.h>
+#include <deal.II/base/logstream.h>
+#include <complex>
+#include <fstream>
+#include <iomanip>
+#include <typeinfo>
+
+template<int rank, int dim,
+         template<int,int,typename> class TensorType,
+         typename NumberType1,
+         typename NumberType2>
+void
+test_tensor_constructor ()
+{
+  deallog
+      << "Rank " << rank << ", "
+      << "Dim " << dim << ":"
+      << "  From " << typeid(NumberType2).name()
+      << "  To " << typeid(NumberType1).name()
+      << " ... "
+      << std::flush;
+  TensorType<rank,dim,NumberType2> tmp2;
+  TensorType<rank,dim,NumberType1> tmp1 (tmp2);
+  deallog << "OK" << std::endl;
+}
+
+template<int rank, int dim,
+         template<int,int,typename> class TensorType,
+         typename NumberType1>
+void
+test_fixed_NT_2()
+{
+  test_tensor_constructor<rank,dim,TensorType,NumberType1,float>();
+  test_tensor_constructor<rank,dim,TensorType,NumberType1,double>();
+  // test_tensor_constructor<rank,dim,TensorType,NumberType1,std::complex<double> >();
```

Comment:
Nope, I did that on purpose. Its a reminder (for me) that one can't convert from a complex<something> to a float/double.